src/System.IdentityModel.Tokens.Jwt/JwtHeader.cs

```diff
@@ -135,6 +172,43 @@ public JwtHeader(EncryptingCredentials encryptingCredentials, IDictionary<string
             EncryptingCredentials = encryptingCredentials;
         }
 
+        /// <summary>
+        /// Initializes a new instance of <see cref="JwtHeader"/>.
+        /// With the Header Parameters:
+        /// <para>{ { typ, JWT }, { alg, SigningCredentials.Algorithm } }</para>
+        /// </summary>
+        /// <param name="encryptingCredentials"><see cref="EncryptingCredentials"/> used when creating a JWS Compact JSON.</param>
+        /// <param name="outboundAlgorithmMap">provides a mapping for the 'alg' value so that values are within the JWT namespace.</param>
+        /// <param name="tokenType"> provides the token type</param>
+        /// <exception cref="ArgumentNullException">If 'signingCredentials' is null.</exception>
+        public JwtHeader(EncryptingCredentials encryptingCredentials, IDictionary<string, string> outboundAlgorithmMap, string tokenType)
+            : base(StringComparer.Ordinal)
+        {
+            if (encryptingCredentials == null)
+                throw LogHelper.LogArgumentNullException(nameof(encryptingCredentials));
+
+            string outboundAlg;
+            if (outboundAlgorithmMap != null && outboundAlgorithmMap.TryGetValue(encryptingCredentials.Alg, out outboundAlg))
+                Alg = outboundAlg;
+            else
+                Alg = encryptingCredentials.Alg;
+
+            if (outboundAlgorithmMap != null && outboundAlgorithmMap.TryGetValue(encryptingCredentials.Enc, out outboundAlg))
+                Enc = outboundAlg;
+            else
+                Enc = encryptingCredentials.Enc;
+
+            if (!string.IsNullOrEmpty(encryptingCredentials.Key.KeyId))
+                Kid = encryptingCredentials.Key.KeyId;
+
+            if (string.IsNullOrEmpty(tokenType))
```

Comment:
similar comment as above. Only thing that looks different is the logic to set Typ

Reply:
That sounds reasonable :)